provider_test.go

```diff
@@ -67,15 +67,15 @@ func TestProviderTypeGetProviderAutodetect(t *testing.T) {
 			t.Setenv("DOCKER_HOST", tt.DockerHost)
 
 			got, err := tt.tr.GetProvider()
-			if (err != nil) != tt.wantErr {
-				t.Errorf("ProviderType.GetProvider() error = %v, wantErr %v", err, tt.wantErr)
+			if tt.wantErr {
+				require.Errorf(t, err, "ProviderType.GetProvider()")
 				return
+			} else {
+				require.NoErrorf(t, err, "ProviderType.GetProvider()")
```

Comment:
suggestion: no need for the else case due to the return, however `wantErr` is never set to true so that can be removed entirely so this becomes just a NoError check.

Sorry for not spotting that before, context was limited in the review display.